### PR TITLE
fix(cdk): share a single subscription to the underlying source observable between multiple subscribers

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Many events are generated every second

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/cdk/services/resize.service.ts [hunk 1]
```diff
@@ -28,25 +26,26 @@ export class TuiResizeService extends ResizeObserverService {
         @Self() @Inject(TuiDestroyService) destroy$: Observable<void>,
         @Inject(RESIZE_OBSERVER_SUPPORT) support: boolean,
         @Inject(RESIZE_OPTION_BOX) box: ResizeObserverBoxOptions,
-        @Inject(ANIMATION_FRAME) animationFrame$: Observable<number>,
     ) {
         super(elementRef, ngZone, support, box);
 
         return this.pipe(
-            catchError(() =>
```
Comment (waterplea) [inline projects/cdk/services/resize.service.ts:36]: `catchError` with a fallback to `requestAnimationFrame` is only here for browsers that do not support `ResizeObserver`. It is necessary to keep that part.

**Review by waterplea — CHANGES_REQUESTED** [review]

File: projects/cdk/services/resize.service.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/services/resize.service.ts:36]: image

do we need support old browsers?

Comment (splincode) [reply projects/cdk/services/resize.service.ts:36]: fixed

Comment (waterplea) [reply projects/cdk/services/resize.service.ts:36]: Yes, we moved support up to Safari 12.1 for `IntersectionObserver` in 3.0.0, but `ResizeObserver` requires Safari 13+.

**Comment by splincode** [comment]
@waterplea hello, please review me)

**Review by waterplea — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**